new/postinstall.sh

```diff
@@ -27,6 +29,19 @@ do
 	mysql -u patchdemo -ppatchdemo patchdemo_$NAME < $PATCHDEMO/sql-perwiki/$sql
 done
 
+# OOUI build
+if [ -d $PATCHDEMO/wikis/$NAME/w/build/ooui ]; then
+	cd $PATCHDEMO/wikis/$NAME/w/build/ooui
+	npm install
+	npm x grunt build
+	cd $PATCHDEMO
+	# JS & CSS
+	cp -r $PATCHDEMO/wikis/$NAME/w/build/ooui/dist/* $PATCHDEMO/wikis/$NAME/w/resources/lib/ooui/
+	# PHP
+	cp -r $PATCHDEMO/wikis/$NAME/w/build/ooui/php/*  $PATCHDEMO/wikis/$NAME/w/vendor/oojs/oojs-ui/php/
```

Comment:
This may be incompatible with the current non-PSR4 autoloading that OOUI-php does.

E.g. https://github.com/wikimedia/mediawiki-vendor/blob/a6e5ad22397e5f304a292fa8bb5e7ea87902c86e/composer/autoload_static.php#L2150

I don't know if it affected patchdemo, maybe it doesn't use composer with a static autoloader.

Reply:
Do you mean when testing a patch that added/removed PHP files?

Reply:
Yes, or some other internal change that might affect how classes are loaded, inherited, traits, composed etc. Doesn't have to be a new widget per-se (which I'm guessing is less common of a change to test as it wouldn't be used).

Reply:
The majority of changes we want to test are JS/CSS only, so I'd be happy to merge this with some warning that PHP changes may not work. If you have any ideas how to fix it, I'm not familiar with the autoload stuff.

Reply:
Ack. Maybe it's better to leave this line off then. Although if it doesn't break anything at least in the common case, I don't mind it either way.

Reply:
It looks like you can use `type:path` in your `repositories` list, and point that at the `build/ooui` folder we just created.

Reply:
Fixed. It's a little slower than a plain copy, but more robust.